### PR TITLE
With buffers

### PR DESCRIPTION
**Pull request by UnixJunkie**

### CODE REVIEW

File: lib/LZ4.ml [hunk 1]
```diff
@@ -50,21 +79,46 @@ module Bigbytes = struct
...
+    else
+      length'
+
   let decompress ~length input =
     if length < 0 then invalid_arg "LZ4.decompress";
     let output  = Array1.create char c_layout length in
-    let length' = C.ba_decompress (bigarray_start array1 input) (bigarray_start array1 output)
-                                  (Array1.dim input) length in
+    let length' = decompress_buff input output 0 length in
     if length' < 0 then
```
Comment (whitequark) [inline lib/LZ4.ml]: No need to perform error checking twice.

File: lib/LZ4.ml [hunk 2]
```diff
@@ -18,30 +18,59 @@ let compress_bound sz =
...
+      length'
+
   let decompress ~length input =
     if length < 0 then invalid_arg "LZ4.decompress";
-    let output  = Bytes.create length in
-    let length' = C.b_decompress (ocaml_bytes_start input) (ocaml_bytes_start output)
-                                 (Bytes.length input) length in
+    let output = Bytes.create length in
+    let length' = decompress_buff input output 0 length in
     if length' < 0 then
```
Comment (whitequark) [inline lib/LZ4.ml]: No need to perform error checking twice.

File: lib/LZ4.ml [hunk 3]
```diff
@@ -18,30 +18,59 @@ let compress_bound sz =
 module type S = sig
   type storage
 
-  val compress   : storage -> storage
+  val compress : storage -> storage
+  val compress_buff : storage -> storage -> offset:int -> length:int -> int
```
Comment (whitequark) [inline lib/LZ4.ml]: I would rather see the `_buff` suffix replaced with `_at`.

**Comment by UnixJunkie** [comment]
I think for very small fixes, it would be faster if you correct them yourself.

**Comment by whitequark** [comment]
There is no rush...

**Comment by UnixJunkie** [comment]
your test suite won't pass if I remove parameter tests you said are extra

**Comment by UnixJunkie** [comment]
the renaming was done

**Pull request merged**